internal/local/structure.go

```diff
@@ -0,0 +1,222 @@
+	if ok, err := fileExists(rootDir); err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, nil
+	}
+
+	var out []map[string]interface{}
+
+	dw := directoryWalker{path: rootDir, onlyDirs: true}
+	if walkErr := dw.walk(func(file os.FileInfo, path string) error {
+		if strings.Contains(path, nameNodeModules) {
+			return nil // skip node_modules
+		}
+
+		config, configErr := parseJSONConfig(path)
+		if configErr != nil {
+			return configErr
+		}
+
+		src, srcErr := parseJavascriptSource(path)
+		if srcErr != nil {
+			return srcErr
+		}
+
+		out = append(out, map[string]interface{}{
+			NameConfig: config,
+			NameSource: src,
+		})
+		return nil
+	}); walkErr != nil {
+		return nil, walkErr
+	}
+
+	return out, nil
+}
+
+func parseGraphQL(rootDir string) (GraphQLStructure, bool, error) {
+	dir := filepath.Join(rootDir, NameGraphQL)
+
+	if ok, err := fileExists(dir); err != nil {
+		return GraphQLStructure{}, false, err
+	} else if !ok {
+		return GraphQLStructure{}, false, nil
+	}
+
+	config, configErr := parseJSONConfig(dir)
+	if configErr != nil {
+		return GraphQLStructure{}, false, configErr
+	}
+
+	customResolvers, customResolversErr := parseJSONFiles(filepath.Join(dir, NameCustomResolvers))
+	if customResolversErr != nil {
+		return GraphQLStructure{}, false, customResolversErr
+	}
+
+	return GraphQLStructure{config, customResolvers}, true, nil
+}
+
+func parseJavascript(rootDir string, file File) (string, error) {
+	src, err := ioutil.ReadFile(filepath.Join(rootDir, file.String()))
+	if err != nil {
+		return "", err
+	}
+	return string(src), nil
+}
+
+func parseJavascriptSource(rootDir string) (string, error) {
+	return parseJavascript(rootDir, FileSource)
+}
+
+func parseJSON(path string) (map[string]interface{}, error) {
+	data, dataErr := readFile(path)
+	if dataErr != nil {
+		return nil, dataErr
+	}
+
+	var out map[string]interface{}
+	if err := unmarshalJSON(data, &out); err != nil {
+		return nil, err
+	}
+	return out, nil
+}
+
+func parseJSONConfig(rootDir string) (map[string]interface{}, error) {
+	return parseJSON(filepath.Join(rootDir, FileConfig.String()))
+}
+
+func parseJSONFiles(rootDir string) ([]map[string]interface{}, error) {
+	if ok, err := fileExists(rootDir); err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, nil
+	}
+
+	out := make([]map[string]interface{}, 0)
+
+	dw := directoryWalker{path: rootDir, onlyFiles: true}
+	if walkErr := dw.walk(func(file os.FileInfo, path string) error {
+		o, err := parseJSON(path)
+		if err != nil {
+			return err
+		}
+
+		out = append(out, o)
+		return nil
+	}); walkErr != nil {
+		return nil, walkErr
+	}
+
+	return out, nil
+}
+
+func parseJSONSecrets(rootDir string) (*SecretsStructure, error) {
+	path := filepath.Join(rootDir, FileSecrets.String())
+
+	if ok, err := fileExists(path); err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, nil // if secrets.json does not exist, continue
+	}
```

Comment:
wouldn't fileExists throw a FileNotFound error?

Reply:
not if im understanding the implementation correctly:

```go
func fileExists(path string) (bool, error) {
	_, err := os.Stat(path)
	if err == nil {
		return true, nil
	}
	if os.IsNotExist(err) {
		return false, nil
	}
	return false, err
}
```

the motivation for even creating this helper was simply to flatten out the error handling from `os.Stat` so there wasn't that nested check for `os.IsNotExist`

Reply:
Seems like we just replaced one check with another though, no? We actually double up the check by using this helper since the helper func checks the IsNotExists error.

`os.Stat -> error -> IsNotExist (true) -> outer func checks error (nil) -> outer func checks ok (false) -> on to next step`

where it could just become 

`os.Stat -> error -> IsNotExist (true) -> on to next step`

This has been in there so I'm fine without the refactor in this PR, I just wanted to call it out now that I've thought it through more.